src/endpoint/s3/s3_errors.js

```diff
@@ -512,6 +517,16 @@ S3Error.InvalidEncodingType = Object.freeze({
     http_code: 400,
 });
 
+////////////////////////////////////////////////////////////////
+// S3 Select not compiled                                     //
+////////////////////////////////////////////////////////////////
+
+S3Error.S3SelectNotCompiled = Object.freeze({
+    code: 'S3 Select was not compiled.',
```

Comment:
error codes are CamelCase by existing standard so change to something like
```suggestion
    code: 'S3SelectNotCompiled',
```